src/ConnectionPool.jl

```diff
@@ -255,10 +299,13 @@ Signal that an entire Request Message has been written to the `Transaction`.
 function IOExtras.closewrite(t::Transaction)
     @require iswritable(t)
 
-    t.c.writebusy = false
-    t.c.writecount += 1                           ;@debug 2 "🗣  Write done: $t"
-    notify(t.c.writedone)
-    notify(poolcondition[Threads.threadid()])
+    @v1_3 lock(t.c.writelock)
+    t.writebusy[] = false
+    Threads.atomic_add!(t.c.writecount, 1)          ;@debug 2 "🗣  Write done: $t"
+    t.c.writebusy[] = false
+    notify(t.c.writelock)
+    @v1_3 unlock(t.c.writelock)
+    release(t.c)
 
     @ensure !iswritable(t)
```

Comment:
This is false now, as you've already released the writelock on the object (the `unlock` call a few lines earlier), so you cannot safely look at any fields now